internal/host/agent_server_http/main_linux.go

```diff
@@ -339,37 +350,45 @@ func PostShutdownFn(ctx context.Context) func(http.ResponseWriter, *http.Request
 func main() {
 	os.Remove(os.Args[0])
 
-	ctx := context.Background()
-
```

Comment:
it is wrong to pass this context to all `Get*Fn(ctx)` below: the context for the returned function **must** come from the HTTP request.